src/jit/lsra.cpp

```diff
@@ -3517,6 +3517,15 @@ static int ComputeOperandDstCount(GenTree* operand)
         // pointers to argument setup stores.
         return 0;
     }
+#ifdef _TARGET_ARMARCH_
+    else if (operand->OperIsPutArgStk())
```

Comment:
Isn't it better to find why srcCount of PUTARG_STK is 0 instead of adding this case? I think setting of srcCount for PUTARG_STK has some problem on ARMARCH.

Reply:
In this method, we are only interested in operands that either produce a register, or are contained. In the case of PUTARG_STK, it may have a srcCount of zero if its child is a `GT_OBJ` with a `GT_LCL_VAR_ADDR` child (see https://github.com/dotnet/coreclr/blob/master/src/jit/lower.cpp#L868)

Reply:
Is it correct on lsra operation when we set srcCount for PUTARG_STK as 0? In x64/x86, srcCount is set by using `GetOperandSourceCount()` (https://github.com/dotnet/coreclr/blob/master/src/jit/lsraxarch.cpp#L1635), but In arm32/arm64, srcCount is set by using only child's source (https://github.com/dotnet/coreclr/blob/master/src/jit/lsraarmarch.cpp#L670)

Reply:
@CarolEidt This can address assertion #13565. However I also have a same question @hseok-oh had. 

Here is an exmaple from #13565. 
Before #13198, `GT_PUTARG_STK` had one srcCount and consume one register during LSRA.
```
N4533 (???,???) [043750] ----GO-------             *  PUTARG_STK [+0x08] stru|tmp550      [+4] $cbe
ct REG NA
+<TreeNodeInfo @ 4533 0=1 1i 0f src=[r0-r9 r12 lr] int=[r0-r9 r12 lr] dst=[r0-r9 r12 lr] I> 
..(suppressed)..
N4533.                    PUTARG_STK [+0x08]
consume=1 produce=0
```

Now,  `GT_PUTARG_STK` has zero srcCount and raises assertion during LSRA.
```
N4533 (???,???) [043748] ----GO-------             *  PUTARG_STK [+0x08] struct REG NA
+<TreeNodeInfo @ 4533 0=0 1i 0f src=[r0-r9 r12 lr] int=[r0-r9 r12 lr] dst=[r0-r9 r12 lr] I>`
...
// and assertion occured
...

Reply:
From the dump you provided, I can't tell what the child of the `PUTARG_STK` node is. Can you tell me which test this is from?

Reply:
If the child is a lclVar, there should be no source register; each word of the struct would be loaded from its stack location into the temp register (note the "1i" in the TreeNodeInfo which indicates that 1 internal register is used). Then it will be stored to the outgoing argument stack location.

Reply:
I'm somewhat late to this comment.  
It seems that `PUTARG_STK` in above example has `GT_ADD` and  as a child.
```
               N003 (  1,  1) [022473] -------------    t22473 =    LCL_VAR   ref    V01 loc1         u:3 $2c0
               N004 (  1,  1) [025101] -------------    t25101 =    CNS_INT   int    56 field offset Fseq[m_cl_op2] $f50
                                                                 /--*  t22473 ref    
                                                                 +--*  t25101 int    
               N005 (  3,  3) [025102] -------------    t25102 = *  ADD       byref  $1044
                                                                 /--*  t25102 byref  
               N006 (  9,  7) [022477] ----GO-------    t22477 = *  OBJ(16)   struct <l:$896, c:$d04>
                                                                 /--*  t22477 struct 
                              [043750] ----GO-------             *  PUTARG_STK [+0x08] struct

```
What do you think of this case ?